src/V4Router.sol

```diff
@@ -42,62 +42,71 @@ abstract contract V4Router is IV4Router, CLRouterBase, BinRouterBase, BaseAction
             if (action == Actions.CL_SWAP_EXACT_IN) {
                 IV4Router.CLSwapExactInputParams calldata swapParams = params.decodeCLSwapExactInParams();
                 _swapExactInput(swapParams);
+                return;
```

Comment:
created https://github.com/pancakeswap/pancake-v4-universal-router/issues/8 to track, maybe this optimisation is applicable for universal router too